testnet/stacks-node/src/syncctl.rs

```diff
@@ -76,19 +79,25 @@ impl PoxSyncWatchdogComms {
     }
 
     /// Wait for at least one inv-sync state-machine passes
-    pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> bool {
+    pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> Result<bool, burnchain_error> {
         let current = self.get_inv_sync_passes();
 
-        let now = get_epoch_time_secs();
         while current >= self.get_inv_sync_passes() {
-            if now + timeout < get_epoch_time_secs() {
-                debug!("PoX watchdog comms: timed out waiting for one inv sync pass");
-                return false;
-            }
-            sleep_ms(1000);
+            self.interruptable_sleep(timeout)?;
             std::sync::atomic::spin_loop_hint();
         }
-        return true;
+        return Ok(true);
+    }
+
+    fn interruptable_sleep(&self, secs: u64) -> Result<(), burnchain_error> {
+        let deadline = secs + get_epoch_time_secs();
+        while get_epoch_time_secs() < deadline {
+            sleep_ms(1000);
+            if !self.should_keep_running.load(Ordering::SeqCst) {
+                return Err(burnchain_error::CoordinatorClosed);
+            }
+        }
+        Ok(())
     }
 
     pub fn wait_for_download_pass(&self, timeout: u64) -> bool {
```

Comment:
You're also going to need to patch `wait_for_download_pass()` to behave like `wait_for_inv_sync_pass()`.

Reply:
Sure. I can see that this function is returning a `bool`, never evaluated. Is this a bug?

Reply:
No, it's fine if you remove the `bool`.

Reply:
Cool, addressed with https://github.com/blockstack/stacks-blockchain/pull/2547/commits/7354a9ca4677aea340d4e538eee5aa38f645a66f then